src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,219 @@
+        String query = "INSERT INTO cars_drivers (driver_id, car_id) VALUES (?,?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement
+                        = connection.prepareStatement(query)) {
+            for (Driver d : drivers) {
+                statement.setLong(1, d.getId());
+                statement.setLong(2, id);
+                statement.executeUpdate();
+            }
+
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't add new relation "
+                    + drivers + " in cars_drivers DB.", e);
+        }
+    }
+
+    private boolean deleteCarDriversRelations(Long id) {
+        String query = "DELETE FROM cars_drivers WHERE car_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            return statement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't delete relations for car with id " + id, e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE cars SET is_deleted = TRUE WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            return statement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't delete car with id " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllCarsByDriver(Long driverId) {
+        List<Car> cars = new ArrayList<>();
+        String getAllCarsByDriverQuery = "SELECT c.id AS car_id, m.id AS manufacturer_id, model, "
+                + "m.name AS manufacturer_name, country "
+                + "FROM cars c "
+                + "JOIN manufacturers m "
+                + "ON c.manufacturer_id = m.id "
+                + "JOIN cars_drivers AS cd "
+                + "ON c.id = cd.car_id "
+                + "WHERE cd.driver_id = ? AND c.is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection
+                        .prepareStatement(getAllCarsByDriverQuery)) {
+            statement.setLong(1, driverId);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                cars.add(getCar(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get car id by driver id " + driverId, e);
+        }
+        if (!cars.isEmpty()) {
+            cars.forEach(c -> c.setDrivers(getDriversForCar(c.getId())));
+        }
```

Comment:
```suggestion
            cars.forEach(c -> c.setDrivers(getDriversForCar(c.getId())));
```
if cars empty, forEach will do nothing